asmcomp/amd64/emit.mlp

```diff
@@ -924,6 +938,27 @@ let emit_instr fallthrough i =
           I.pop r11;
           I.jmp r11
       end
+  | Lop(Ipollcall { check_young_limit = do_check }) ->
```

Comment:
```suggestion
  | Lop(Ipollcall { check_young_limit }) ->
```
You can elide having to explicitly name variables.

Reply:
What is the purpose of `check_young_limit`? If the young_limit is not checked, is it the same as not having the poll point?

Reply:
The `ipollcall` can check the young limit itself (for places like the function prologue or when we raise exceptions) but it can also just assume the check has already been done and unconditionally call `caml_poll`.

The second case is where it's used at the back edge of a loop because it's inside an ifthenelse which itself checks the young limit.

It's a little awkward having the same op do both but the alternative is to have two separate ops and this leads to a load of code duplication in anything that touches Mach and the emitter.

Reply:
Ok